snapcraft/cli.py

```diff
@@ -34,31 +36,61 @@
 from . import commands
 from .legacy_cli import run_legacy
 
+
+@dataclass
+class CommandGroup:
+    """Dataclass to hold a command group."""
+
+    name: str
+    commands: list
+
+
+CORE22_LIFECYCLE_COMMAND_GROUP = CommandGroup(
+    "Lifecycle",
+    [
+        commands.core22.CleanCommand,
+        commands.core22.PullCommand,
+        commands.core22.BuildCommand,
+        commands.core22.StageCommand,
+        commands.core22.PrimeCommand,
+        commands.core22.PackCommand,
+        commands.core22.SnapCommand,  # hidden (legacy compatibility)
+        commands.core22.TryCommand,
+    ],
+)
+
+CORE24_LIFECYCLE_COMMAND_GROUP = CommandGroup(
+    "Lifecycle",
+    [
+        craft_application.commands.lifecycle.CleanCommand,
+        craft_application.commands.lifecycle.PullCommand,
+        craft_application.commands.lifecycle.BuildCommand,
+        craft_application.commands.lifecycle.StageCommand,
+        craft_application.commands.lifecycle.PrimeCommand,
+        commands.PackCommand,
+        commands.SnapCommand,  # Hidden (legacy compatibility)
+        commands.RemoteBuildCommand,
+        commands.TryCommand,
```

Comment:
Are we supporting the try command with core24?

Reply:
Not yet.  It will raise a FeatureNotImplemented error later on.

`try` was the last unimplemented command and we know it will be a child of the Lifecycle command class, so I moved it there and removed the `Unimplemented` mixin class.